### DISCUSSION TITLE
Wiki for PatternFly Elements?

### DISCUSSION THREAD

**Discussion by bennypowers** [patternfly/patternfly-elements#2428] [Ideas]
@anujsingla suggested we add a document outlining our codebase to the repo, inspired by this one from salesforce design system: https://github.com/salesforce/design-system-react/blob/master/docs/codebase-overview.md

Should we add a Wiki to this repo, similar to the one we have on RHDS? Most of the content would be the same or similar, or we could like to the RHDS wiki for things like code standards, while outlining the repository structure, FAQs, etc.

**Comment by brianferry** [comment]
I think it would make sense to have a wiki over here, if only to point to our RHDS wiki if we'd rather only update one area.  That way if someone comes into this repo and isn't aware of the RHDS one they can still find the information.

**Comment by bennypowers** [comment]
@anujsi I'll start a wiki so you can add contributing docs, resources, etc.

Make sure to link to the [RHDS wiki](https://github.com/RedHat-UX/red-hat-design-system/wiki)

and while we're here, here's the snippet from earlier today

```html
<script type="module">
import 'https://jspm.dev/@patternfly/elements';
const modal = document.querySelector('pf-modal');
modal.addEventListener('click', function(event) {
  if (event.target.localName == 'pf-button') {
    const button = event.target;
    modal.close(button.dataset.action);
  }
});

modal.addEventListener('close', function(event) {
  switch (modal.returnValue) {
    case 'confirm':
      // fetch(....)
      console.log('deleted')
      break;
    default:
      console.log('cancel');
      break;
  }
});
</script>

<pf-button id="confirm-delete">Delete</pf-button>
<pf-modal trigger="confirm-delete">
  <p>Are you sure?</p>
  <pf-button slot="footer" data-action="confirm" danger>Yes</pf-button>
  <pf-button slot="footer" data-action="cancel">No</pf-button>
</pf-modal>
```

**Reply by bennypowers** [reply]
https://github.com/patternfly/patternfly-elements/wiki
